pkg/platform/k8s/reconciler.go

```diff
@@ -54,31 +54,30 @@ func (r *Reconciler) CreateResource(name string, obj client.Object) {
 
 // UpdateResource updates k8s resource.
 func (r *Reconciler) UpdateResource(name string, obj client.Object) {
-	if oldObj, ok := r.list[name]; ok {
-		if obj.GetName() == oldObj.GetName() {
-			err := r.client.Update(context.Background(), obj)
-			if err != nil {
-				r.logger.Errorf("error occurred while updating K8s %v %v:", reflect.TypeOf(obj).String(), err)
-				r.failedList[name] = reconcileObj{spec: obj, op: updateOp}
-				return
-			}
-			r.list[name] = obj
+	oldObj, ok := r.list[name]
+	if !ok {
+		r.logger.Errorf("error occurred while updating K8s %v: %v is not exist", reflect.TypeOf(obj).String(), name)
+	}
+
+	if obj.GetName() == oldObj.GetName() {
+		err := r.client.Update(context.Background(), obj)
+		if err != nil {
+			r.logger.Errorf("error occurred while updating K8s %v %v:", reflect.TypeOf(obj).String(), err)
+			r.failedList[name] = reconcileObj{spec: obj, op: updateOp}
 			return
```

Comment:
Why did you move the return inside the error case?

Reply:
should processing (create resource) continue when the Update failed? Does it depend on the error returned?
Perhaps I misunderstood the logic... Let me try to revert this portion by removing the return and see if tests pass